autogpt_platform/backend/backend/util/service.py

```diff
@@ -190,11 +191,10 @@ def set_log_level(self, log_level: str):
     def _handle_internal_http_error(status_code: int = 500, log_error: bool = True):
         def handler(request: Request, exc: Exception):
             if log_error:
-                if status_code == 500:
-                    log = logger.exception
-                else:
-                    log = logger.error
-                log(f"{request.method} {request.url.path} failed: {exc}")
+                logger.error(
+                    f"{request.method} {request.url.path} failed: {exc}",
+                    exc_info=exc if status_code == 500 else None,
+                )
```

Comment:
is the stacktrace for unkown error not useful ?

Reply:
Yes, because 400 errors mean invalid input while 500 errors mean internal error. If you check the current logic it has the same condition except it uses `logger.exception` which doesn't work in this context.

Reply:
ah got it :)